### PR TITLE
fallback LDAP to standard password-based login

### PR DESCRIPTION
**Pull request by andrzejwp**
Hello,

I'm submitting a very simple fallback mechanism that supports LDAP as a primary provider and falls back to the standard method in case a user is not found in LDAP.

This is a rough sketch we've been using with our instance for a couple of months now and serves the following bussiness case - our company (CodeWave) has an internal LDAP server with our employees data in it. Rocket.Chat, as well as other internal services, are bound to our LDAP and so our employees sign-in easily with the same credentials. At the same time we need to provide access to some of our resources to our clients and so - we usually need separate login methods for them to access our bugtracker / chat / etc. The fallback mechanism supports such a case by allowing internal users to login directly through LDAP credentials and providing standard registration / authentication for our clients.

### CODE REVIEW

**Comment by engelgabriel** [comment]
lgtm

File: packages/rocketchat-ui-login/login/form.coffee [hunk 1]
```diff
@@ -112,15 +112,16 @@ Template.loginForm.events
...
 				Meteor[loginMethod] formData.emailOrUsername, formData.pass, (error) ->
 					RocketChat.Button.reset(button)
-					if error?
-						if error.error is 'no-valid-email'
-							instance.state.set 'email-verification'
-						else
-							toastr.error t 'User_not_found_or_incorrect_password'
-						return
+					loginMethod = 'loginWithPassword'
+					Meteor[loginMethod] formData.emailOrUsername, formData.pass, (error) ->
```
Comment (sampaiodiego) [inline packages/rocketchat-ui-login/login/form.coffee]: this code is always calling *both* login methods, even if the first one (LDAP) succed..
I think the fallback call should be done only if the first returns an error..

**Comment by engelgabriel** [comment]
@rodrigok was suggesting an idea to handle this on the server side, would that be preferable?

**Comment by andrzejwp** [comment]
Yes, I guess doing it on the server side would be the logical way to do it, along with a configurable chain of auth providers. That's beyond the level of Coffee I can ingest, unfortunately ;) I'll happily back to it when you guys move to ES6 (as my +1 to your plans).

**Comment by andrzejwp** [comment]
What would you expect me to do with the code in order to merge it?

**Comment by engelgabriel** [comment]
Let me talk to the @RocketChat/core and get back to you.

**Comment by engelgabriel** [comment]
Close in favor of https://github.com/RocketChat/Rocket.Chat/pull/1957

**Pull request closed without merging**